collector/utils.go

```diff
@@ -157,6 +170,54 @@ func sortTaskRunsForGapCalculations(pr *v1.PipelineRun, oc client.Client, ctx co
 	return sortedTaskRunsByCreateTimes, reverseOrderSortedTaskRunsByCompletionTimes, false
 }
 
+func tagPipelineRunsWithTaskRunsGettingThrottled(pr *v1.PipelineRun, oc client.Client, ctx context.Context) error {
+	throttled := false
+	throttledTaskRun := ""
+	for _, kidRef := range pr.Status.ChildReferences {
+		if kidRef.Kind != "TaskRun" {
+			continue
+		}
+		kid := &v1.TaskRun{}
+		err := oc.Get(ctx, types.NamespacedName{Namespace: pr.Namespace, Name: kidRef.Name}, kid)
+		if err != nil && !errors.IsNotFound(err) {
+			ctrl.Log.Info(fmt.Sprintf("could get taskrun %s:%s: %s", pr.Namespace, kidRef.Name, err.Error()))
+			return err
+		}
+		succeedCondition := kid.Status.GetCondition(apis.ConditionSucceeded)
+		if succeedCondition != nil && succeedCondition.Status == corev1.ConditionUnknown {
+			switch succeedCondition.Reason {
+			case pod.ReasonExceededResourceQuota:
+				throttled = true
+				throttledTaskRun = kid.Name
+				break
+			case pod.ReasonExceededNodeResources:
+				throttled = true
+				throttledTaskRun = kid.Name
+				break
+			}
+		}
+	}
+	// for our purposes, labelling only the first throttling instances is sufficient
+	previouslyLabelled := true
+	if pr.Labels == nil {
+		previouslyLabelled = false
+	}
```

Comment:
Can we omit that? Doesn't the next line `_, previouslyLabelled := pr.Labels[THROTTLED_LABEL]` (:=) have the same effect?

Reply:
yeah I should also instantiate a labels if it is nil ... I'll combine that with your suggestion - thanks